shunting-yard-replicator/src/test/java/com/hotels/shunting/yard/replicator/exec/messaging/MessageReaderAdapterTest.java

```diff
@@ -0,0 +1,224 @@
+
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import com.hotels.shunting.yard.common.event.EventType;
+import com.hotels.shunting.yard.common.event.SerializableAddPartitionEvent;
+import com.hotels.shunting.yard.common.event.SerializableAlterPartitionEvent;
+import com.hotels.shunting.yard.common.event.SerializableCreateTableEvent;
+import com.hotels.shunting.yard.common.event.SerializableDropPartitionEvent;
+import com.hotels.shunting.yard.common.event.SerializableDropTableEvent;
+import com.hotels.shunting.yard.common.event.SerializableInsertEvent;
+import com.hotels.shunting.yard.common.event.SerializableListenerEvent;
+import com.hotels.shunting.yard.common.messaging.MessageReader;
+import com.hotels.shunting.yard.replicator.exec.event.MetaStoreEvent;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MessageReaderAdapterTest {
+
+  private static final List<String> PARTITION_VALUES = ImmutableList.of("value_1", "value_2", "value_3");
+  private static final List<String> PARTITION_COLUMNS = ImmutableList.of("Column_1", "Column_2", "Column_3");
+  private static final Map<String, String> EMPTY_MAP = ImmutableMap.of();
+  private static final String TEST_DB = "test_db";
+  private static final String TEST_TABLE = "test_table";
+
+  private MessageReaderAdapter messageReaderAdapter;
+
+  private @Mock SerializableAddPartitionEvent addPartitionEvent;
+  private @Mock SerializableAlterPartitionEvent alterPartitionEvent;
+  private @Mock SerializableDropPartitionEvent dropPartitionEvent;
+  private @Mock SerializableDropTableEvent dropTableEvent;
+  private @Mock SerializableInsertEvent insertEvent;
+  private @Mock SerializableCreateTableEvent createTableEvent;
+  private @Mock MessageReader messageReader;
+  private @Mock Table dummyHiveTable;
+  private @Mock Partition partition;
+
+  private List<Partition> partitionValues;
+  private List<FieldSchema> partitionKeys;
+
+  @Before
+  public void init() {
+    FieldSchema partitionColumn1 = new FieldSchema("Column_1", "String", "");
+    FieldSchema partitionColumn2 = new FieldSchema("Column_2", "String", "");
+    FieldSchema partitionColumn3 = new FieldSchema("Column_3", "String", "");
+
+    partitionKeys = ImmutableList.of(partitionColumn1, partitionColumn2, partitionColumn3);
+    partitionValues = ImmutableList.of(partition);
+    messageReaderAdapter = new MessageReaderAdapter(messageReader);
+    when(partition.getValues()).thenReturn(PARTITION_VALUES);
+    when(dummyHiveTable.getPartitionKeys()).thenReturn(partitionKeys);
+  }
+
+  @After
+  public void close() throws IOException {
```

Comment:
not sure I get this, why close it always afterwards. Better to have a normal test that verifies the underlying reader close maybe?

Reply:
I have added an additional test for close() but not sure why the Adapter should not be closed once the test is complete ?

Reply:
it's all mocks, what you gonna close? :)

Reply:
Yeah. I see what you mean. I guess, I can remove that.